whisper_server.py

```diff
@@ -212,10 +212,13 @@ def load_whisper_model(device='GPU', model_size='small'):
     Load the Whisper model once. Return the model object.
     """
     logging.info(f"Loading Whisper model ({model_size}), device={device}")
-    if device.upper() == "GPU" and torch.cuda.is_available():
-        return whisper.load_model(model_size).to('cuda')
-    else:
-        return whisper.load_model(model_size).to('cpu')
+    if device.upper() == "GPU":
+        if torch.cuda.is_available():
+            return WhisperModel(model_size, device="cuda", compute_type="int8_float16")
+        else:
+            logging.error("cuda not available so using CPU")

```

Comment:
If cuda is not available then log error and fallback on CPU. We should perhaps throw an error here instead, but will need to do after moving to logging to file as otherwise the terminal will be killed and you'll lose the reason why.